### PR TITLE
Check the DB state at boot

### PR DESCRIPTION
**Pull request by jjos2372**

### CODE REVIEW

File: src/brs/BlockchainProcessorImpl.java [hunk 1]
```diff
@@ -764,6 +770,33 @@ private void addBlock(Block block) {
...
+    }
+    for (Escrow escrow : escrowService.getAllEscrowTransactions()) {
+      totalEffectiveBalance += escrow.getAmountNQT();
+    }
+    
+    logger.info("Total mined {}, total effective on wallets {}", totalMined, totalEffectiveBalance);
+    
+    if(totalEffectiveBalance > totalMined) {
+      logger.info("Database is inconsistent, please sync from empty or add 'DB.SkipCheck = true' at your own risk.");
+      System.exit(-1);
```
Comment (ohager) [inline src/brs/BlockchainProcessorImpl.java:797]: awesome!

File: src/brs/http/GetState.java [hunk 2]
```diff
@@ -55,16 +57,25 @@ JsonElement processRequest(HttpServletRequest req) {
 
     if ("true".equalsIgnoreCase(req.getParameter(INCLUDE_COUNTS_PARAMETER))) {
       long totalEffectiveBalance = 0;
+      long totalMined = 0;
+      
+      for (int i=0; i <= blockchain.getHeight(); i++) {
+    	totalMined += BlockServiceImpl.getBlockReward(i);
```
Comment (ohager) [inline src/brs/http/GetState.java:63]: missing indent?

File: src/brs/services/impl/BlockServiceImpl.java [hunk 3]
```diff
@@ -187,12 +187,16 @@ public void apply(Block block) {
...
-    if (block.getHeight() == 0 || block.getHeight() >= 1944000) {
-      return 0;
-    }
-    int month = block.getHeight() / 10800;
-    return BigInteger.valueOf(10000).multiply(BigInteger.valueOf(95).pow(month))
-        .divide(BigInteger.valueOf(100).pow(month)).longValue() * Constants.ONE_BURST;
+    return getBlockReward(block.getHeight());
+  }
+
+  public static long getBlockReward(int height) {
```
Comment (ohager) [inline src/brs/services/impl/BlockServiceImpl.java:193]: not sure why you extracted this here...

**Review by ohager — APPROVED** [review]
LGTM - just that I didn't understand the block reward extraction, but you'll have your reasons :grin:

File: src/brs/services/impl/BlockServiceImpl.java (hunk 3 above)
Comment (jjos2372) [reply src/brs/services/impl/BlockServiceImpl.java:193]: This one because I need the block reward having only the height on another place.

**Review by blankey1337 — APPROVED** [review]

**Pull request merged**